components/camel-jira/src/main/java/org/apache/camel/component/jira/FileConverter.java

```diff
@@ -34,20 +35,13 @@ private FileConverter() {
     @Converter
     public static File genericToFile(GenericFile<File> genericFile, Exchange exchange) throws IOException {
         Object body = genericFile.getBody();
-        File file;
+        File file = null;
+        Path path;
         if (body instanceof byte[]) {
             byte[] bos = (byte[]) body;
-            String destDir = System.getProperty("java.io.tmpdir");
-            if (destDir != null && !destDir.endsWith(File.separator)) {
-                destDir += File.separator;
-            }
-            file = new File(destDir, genericFile.getFileName());
-            if (!file.getCanonicalPath().startsWith(destDir)) {
-                throw new IOException("File is not jailed to the destination directory");
-            }
-            Files.write(file.toPath(), bos, StandardOpenOption.CREATE);
-            // delete the temporary file on exit, as other routing may need the file for post processing
-            file.deleteOnExit();
+            path = Files.createTempFile(genericFile.getFileName(), null, null);
```

Comment:
Don't pass an argument to the `attrs` varargs. Without it, it will be implicitly the empty array, which is what you want. If you pass values, the default, secure settings won't be used.

```suggestion
            path = Files.createTempFile(genericFile.getFileName(), null);
```

Reply:
Thanks for the hint, didn't notice that.